feedparser/namespaces/georss.py

```diff
@@ -175,12 +175,15 @@ def _parse_poslist(value, geom_type, swap=True, dims=2):
 def _gen_georss_coords(value, swap=True, dims=2):
     # A generator of (lon, lat) pairs from a string of encoded GeoRSS
     # coordinates. Converts to floats and swaps order.
-    latlons = (float(ll) for ll in value.replace(',', ' ').split())
+    latlons = ( float(ll) for ll in value.replace(',', ' ').split() )
     while True:
-        t = [next(latlons), next(latlons)][::swap and -1 or 1]
-        if dims == 3:
-            t.append(next(latlons))
-        yield tuple(t)
+        try:
+            t = [next(latlons), next(latlons)][::swap and -1 or 1]
+            if dims == 3:
+                t.append(next(latlons))
+            yield tuple(t)
+        except StopIteration:
+            return
```

Comment:
What do you think if besides fixing the bug, we make the implementation a bit more clear and robust?

Something like:
```
def _gen_georss_coords(value, swap=True, dims=2):
    """A generator of (lon, lat) pairs from a string of encoded GeoRSS
    coordinates. Converts to floats and swaps order.
    
    Examples
    --------
    >>> list(_gen_georss_coords('1,10,2,20'))
    [(10.0, 1.0), (20.0, 2.0)]
    """
    coords_list = map(float, value.split(','))
    while True:
        coords = list(itertools.islice(coords_list, dims))
        if not len(coords):
            break
        elif len(coords) == dims:
            if swap:
                coords[:2] = reversed(coords[:2])
            yield tuple(coords)
        else:
            raise ValueError(
                'Inconsistent number of dimensions found. '
                "Can't divide {} coordinates into {} dimensions".format(
                    value.count(',') + 1, dims))
```